### PR TITLE
Release - 0.3.0

### PR DESCRIPTION
**Pull request by manh-t**
## Features

- #53 
- #54 
- #55 
- #56 

## Chores

- #52 
- #91

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64c251c664f0420695bf9c34--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/../Dashboard/index.tsx](../blob/17e3285/src/screens/Dashboard/index.tsx) | (1/6) 17% | (1/6) 17% | (0/4) 0% | (0/0) NaN%
[src/../v1/requestManager.ts](../blob/17e3285/src/lib/request/v1/requestManager.ts) | (9/18) 50% | (9/18) 50% | (4/6) 67% | (1/3) 33%
[src/config/index.ts](../blob/17e3285/src/config/index.ts) | (5/9) 56% | (5/9) 56% | (0/3) 0% | (0/4) 0%
[src/tests/error.ts](../blob/17e3285/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/screens/SignIn/index.tsx](../blob/17e3285/src/screens/SignIn/index.tsx) | (12/16) 75% | (12/16) 75% | (3/6) 50% | (8/8) 100%
[src/../Content/index.tsx](../blob/17e3285/src/components/Dashboard/Content/index.tsx) | (8/10) 80% | (8/10) 80% | (4/6) 67% | (3/5) 60%
[src/../index.tsx](../blob/17e3285/src/routes/ProtectedRoute/index.tsx) | (4/5) 80% | (4/5) 80% | (2/2) 100% | (1/2) 50%
[src/App.tsx](../blob/17e3285/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/17e3285/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/17e3285/src/adapters/Authentication/index.ts) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/17e3285/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/17e3285/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/17e3285/src/adapters/Survey/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/components/Alert/index.tsx](../blob/17e3285/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
[src/../index.tsx](../blob/17e3285/src/components/BackgroundImage/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (2/2) 100%
Other (22 more) | (98/98) 100% | (98/98) 100% | (34/34) 100% | (28/29) 97%
Total | (172/198) 87% | (172/198) 87% | (60/75) 80% | (48/60) 80%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a84b65732526a53212d7bda38c5fe1e15dd97de3
</p>

File: src/components/Dashboard/Content/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      id: '1',
+      resourceType: 'survey',
+      coverImageUrl: 'https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_',
+      title: 'Working from home Check-In',
+      description: 'We would like to know how you feel about our work from home.',
+    },
+  ];
+
+  it('renders DashboardContent and its components', () => {
```
Comment (tyrro) [inline src/components/Dashboard/Content/index.test.tsx:18]:
```suggestion
  it('renders Dashboard and its contents', () => {
```

File: src/components/Dashboard/Content/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    );
+
+    const dashboardContent = screen.getByTestId(dashboardDataTestIds.content);
+
+    expect(dashboardContent).toBeVisible();
+    expect(dashboardContent).toHaveTextContent(surveys[0].title);
+    expect(dashboardContent).toHaveTextContent(surveys[0].description);
+  });
+
+  it('does NOT render the DashboardContent components', () => {
```
Comment (tyrro) [inline src/components/Dashboard/Content/index.test.tsx:36]: sorry I missed it previously, currently we have only two test blocks:

```
it('renders Dashboard and its contents', () => {
it('does NOT render the DashboardContent components', () => {
```

could we provide a bit more context here?

```js
describe('when shouldShowShimmer is true', ()  => {
  it('renders Dashboard and its contents', () => {
  
describe('when shouldShowShimmer is false', ()  => {
  it('does NOT render Dashboard and its contents', () => {
```

File: public/index.html [hunk 3]
```diff
@@ -25,7 +25,7 @@
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
+    <div id="root" class="root-container"></div>
```
Comment (nvminhtue) [inline public/index.html]: Please remove this commit

**Review by nvminhtue — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/Header/index.tsx [hunk 4]
```diff
@@ -1,34 +1,49 @@
...
 import { ReactComponent as NimbleLogoWhite } from 'assets/images/icons/nimble-logo-white.svg';
+import Shimmer from 'components/Shimmer';
 
 interface DashboardHeaderProps {
   dateTime: string;
   daysAgo: string;
   profileUrl: string;
+  shouldShowShimmer?: boolean;
   children: React.ReactNode;
   'data-test-id'?: string;
```
Comment (tyrro) [inline src/components/Dashboard/Header/index.tsx:12]: could we move the `data-test-id` inside the component, just like `DashboardContent`?

File: src/components/Dashboard/Header/index.test.tsx [hunk 5]
```diff
@@ -21,10 +21,26 @@ describe('DashboardHeader', () => {
...
     expect(dashboardHeader).toHaveTextContent(dateTime);
-    expect(dashboardHeader).toHaveTextContent(dateTime);
-    expect(dashboardHeader).toHaveTextContent(dateTime);
+    expect(dashboardHeader).toHaveTextContent(daysAgo);
 
     expect(avatar).toBeVisible();
     expect(avatar).toHaveAttribute('src', profileUrl);
   });
+
+  it('does NOT renders text components', () => {
```
Comment (tyrro) [inline src/components/Dashboard/Header/index.test.tsx:30]: could we have the contexts here as well?

```js
describe('when shouldShowShimmer is true', ()  => {
  it('renders text components, () => {
  
describe('when shouldShowShimmer is false', ()  => {
  it('does NOT render text components', () => {
```

**Review by tyrro — APPROVED** [review]
Small suggestions that I missed previously, apart from them LGTM!

File: public/index.html (hunk 3 above)
Comment (manh-t) [reply public/index.html]: @nvminhtue thanks, I have rebased again to remove this commit

File: src/components/Dashboard/Content/index.test.tsx (hunk 1 above)
Comment (manh-t) [reply src/components/Dashboard/Content/index.test.tsx:18]: @tyrro I don't want to edit the release PR, so I will create a ticket to fix these comments. Please ignore them for now 🙇

Comment (manh-t) [reply src/components/Dashboard/Content/index.test.tsx:18]: @tyrro created it https://github.com/manh-t/react-survey/issues/92 ✅

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**